server/prisma-rs/libs/database-introspection/tests/serialization_tests.rs

```diff
@@ -0,0 +1,147 @@
+#![allow(non_snake_case)]
+#![allow(unused)]
+
+use barrel::{types, Migration};
+use database_introspection::*;
+use log::{debug, LevelFilter};
+use pretty_assertions::assert_eq;
+use prisma_query::connector::{Queryable, Sqlite as SqliteDatabaseClient};
+use std::collections::HashSet;
+use std::fs::File;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+use std::{thread, time};
+
+const SCHEMA: &str = "DatabaseInspectorTest";
+
+static IS_SETUP: AtomicBool = AtomicBool::new(false);
+
+fn setup() {
+    let is_setup = IS_SETUP.load(Ordering::Relaxed);
+    if is_setup {
+        return;
+    }
+
+    let log_level = match std::env::var("RUST_LOG")
+        .unwrap_or("warn".to_string())
+        .to_lowercase()
+        .as_ref()
+    {
+        "trace" => LevelFilter::Trace,
+        "debug" => LevelFilter::Debug,
+        "info" => LevelFilter::Info,
+        "warn" => LevelFilter::Warn,
+        "error" => LevelFilter::Error,
+        _ => LevelFilter::Warn,
+    };
+    fern::Dispatch::new()
+        .format(|out, message, record| {
+            out.finish(format_args!("[{}][{}] {}", record.target(), record.level(), message))
+        })
+        .level(log_level)
+        .chain(std::io::stdout())
+        .apply()
+        .expect("fern configuration");
+
+    IS_SETUP.store(true, Ordering::Relaxed);
+}
+
+#[test]
+fn database_schema_is_serializable() {
```

Comment:
This test is the direction i imagined. It would be great if we could add more cases though. The following things caught my eye:

- The test doesn't check whether null for a primary key works.
- The tests should catch when an enum variant is renamed. So we should exercise each enum variant we have.